kuzzleio/kuzzle
# Fix OpenApi Format + Implement route testing

## What does this PR do ?
Fixes two issues in our **OpenApi** Format:

- Seems like there was a breaking change the moment we implemented the `/_/`prefix on our plugin routes. Our generator should not add any prefixes by itself. `/_/_plugin/` & `/_plugin/_plugin/` were found.
- Some plugin **verbs** were getting translated in all caps. Make sure every verb is lowercase.

### How should this be manually tested?
1. Get Kuzzle's OpenApi result in any format. (http://localhost:7512/_openapi[?format=yaml])
2. Copy/Paste into [Swagger Editor](https://editor.swagger.io/)

You should not have any structural issues reported by the editor & no  `/_/_plugin/` || `/_plugin/_plugin/` routes should be found.

### Other changes

- Took the opportunity to **test** our **openApi** route :rescue_worker_helmet: 
- Took the opportunity to add some unit **tests** to our **openApiGenerator** :rescue_worker_helmet: 
- :warning: Changed our **swagger.js** generator file to the more openApi related **openApiGenerator.js** If you need info about why: [SwaggerVsOpenApi](https://swagger.io/blog/api-strategy/difference-between-swagger-and-openapi/)

Closes #1910 Exposed OpenApi is not valid
